thrust/testing/reduce.cu

```diff
@@ -164,8 +165,8 @@ void TestReduceCountingIterator()
 
   ASSERT_LEQUAL(T(n), unittest::truncate_to_max_representable<T>(n));
 
-  thrust::counting_iterator<T, thrust::host_system_tag> h_first   = thrust::make_counting_iterator<T>(0);
-  thrust::counting_iterator<T, thrust::device_system_tag> d_first = thrust::make_counting_iterator<T>(0);
+  cuda::counting_iterator<T> h_first{0};
+  cuda::counting_iterator<T> d_first{0};
```

Comment:
Critical: This is not equivalent, since `thrust::reduce` later has no other iterator to deduce on which system it should run. Can you please change below to:
```c++
  T h_result = thrust::reduce(thrust::host, h_first, h_first + n, init);
  T d_result = thrust::reduce(thrust::device, d_first, d_first + n, init);
```